### PR TITLE
Event updates

### PR DESCRIPTION
**Pull request by rauhul**

### CODE REVIEW

File: HackIllinois/APIService/HIEventService.swift [hunk 1]
```diff
@@ -16,17 +16,17 @@ import CoreData
...
+    static func create(event: HIAPIEvent) -> APIRequest<HIAPIEvent> {
         let eventDict = [String: Any]()
-        return APIRequest<HIAPIEvent.Contained>(service: self, endpoint: "", body: eventDict, method: .POST)
+        return APIRequest<HIAPIEvent>(service: self, endpoint: "", body: eventDict, method: .POST)
     }
 
-    static func getAllEvents(active: Bool = false) -> APIRequest<HIAPIEvent.Contained> {
-        let paramaters = ["active": "\(active)"]
+    static func getAllEvents() -> APIRequest<HIAPIEvent.Contained> {
+        let paramaters = ["Content-Type": "application/json"]
```
Comment (rauhul) [inline HackIllinois/APIService/HIEventService.swift]: why?

File: HackIllinois/Models/APIModels/HIAPIEvent.swift [hunk 2]
```diff
@@ -11,32 +11,37 @@
...
+import APIManager
 
-struct HIAPIEvent: Codable {
-    typealias Contained = HIAPIReturnDataContainer<HIAPIEvent>
+struct HIAPIEventsContainer: Decodable, APIReturnable {
+    let events: [HIAPIEvent]
+}
+
+struct HIAPIEvent: Codable, APIReturnable {
+    typealias Contained = HIAPIEventsContainer
```
Comment (rauhul) [inline HackIllinois/Models/APIModels/HIAPIEvent.swift]: remove this

Comment (rauhul) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: drop APIReturnable if its not needed

File: HackIllinois/Models/APIModels/HIAPIEvent.swift [hunk 3]
```diff
@@ -11,32 +11,37 @@
...
-    var end: Date
-    var start: Date
-    var locations: [HIAPILocationReference]
-
-    struct HIAPILocationReference: Codable {
-        var id: Int16
-        var eventId: Int16
-        var locationId: Int16
-    }
+    var end: Double
```
Comment (rauhul) [inline HackIllinois/Models/APIModels/HIAPIEvent.swift]: why is this a double?

File: HackIllinois/APIService/HIEventService.swift (hunk 1 above)
Comment (darthbatman) [reply HackIllinois/APIService/HIEventService.swift]: the active parameter isn't necessary as far as i know, should work as it was though

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 2 above)
Comment (darthbatman) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: alright

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 3 above)
Comment (darthbatman) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: because TimeInterval accepts a double when converting to Date, but Int/Int16 should also work fine

File: HackIllinois/APIService/HIEventService.swift (hunk 1 above)
Comment (rauhul) [reply HackIllinois/APIService/HIEventService.swift]: no I mean `let paramaters = ["Content-Type": "application/json"]`

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 3 above)
Comment (rauhul) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: What is the number coming from the API? Time since 1970?

File: HackIllinois/APIService/HIEventService.swift (hunk 1 above)
Comment (darthbatman) [reply HackIllinois/APIService/HIEventService.swift]: not explicitly necessary, can be removed

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 3 above)
Comment (rauhul) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: then this should be type `Date` where the APIReturnable `init(from:)` method is overriden to provide a JSONDecoder that uses the proper date decoding strategy, see docs [here](https://developer.apple.com/documentation/foundation/jsondecoder/datedecodingstrategy)

File: HackIllinois/APIService/HIEventService.swift (hunk 1 above)
Comment (rauhul) [reply HackIllinois/APIService/HIEventService.swift]: Just FYI, this data is never encoded as a `HTTPURLParameter`, it should be a `HTTPHeader`, however `HIBaseService` already does this for you.

Comment (darthbatman) [reply HackIllinois/APIService/HIEventService.swift]: ok

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 3 above)
Comment (darthbatman) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: ah ok, currently it loops through apievents and makes them events (which have type `Date`)

File: HackIllinois/Models/APIModels/HIAPIEvent.swift (hunk 2 above)
Comment (darthbatman) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: APIReturnable is necessary, as create returns an HIAPIEvent

Comment (rauhul) [reply HackIllinois/Models/APIModels/HIAPIEvent.swift]: mmk

**Comment by rauhul** [comment]
Resolved the issue above by setting:
``` swift
container.persistentStoreDescriptions.first?.shouldAddStoreAsynchronously = false
```

File: HackIllinois/Models/CoreDataModels/Event+CoreDataClass.swift [hunk 4]
```diff
@@ -16,15 +16,18 @@ import HIAPI
 
 @objc(Event)
 public class Event: NSManagedObject {
-    convenience init(context moc: NSManagedObjectContext, event: HIAPI.Event, locations: NSSet) {
+
+    convenience init(context moc: NSManagedObjectContext, event: HIAPI.Event, locations: NSSet, favorite: Bool) {
```
Comment (rauhul) [inline HackIllinois/Models/CoreDataModels/Event+CoreDataClass.swift]: Remove

File: HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift [hunk 5]
```diff
@@ -77,7 +78,7 @@ extension HIEventCell {
         titleLabel.text = rhs.name
         contentStackViewHeight += titleLabel.intrinsicContentSize.height
         lhs.contentStackView.addArrangedSubview(titleLabel)
-        if let locations = rhs.locations as? Set<Location> {
+        if let locations = rhs.locations.allObjects as? [Location] {
```
Comment (rauhul) [inline HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift]: Revert, how to ensure sorted properly?

File: HackIllinois/ViewControllers/HIEventDetailViewController.swift [hunk 6]
```diff
@@ -35,24 +35,24 @@ extension HIEventDetailViewController {
...
-                    case .failure(let error):
-                        print(error, error.localizedDescription)
+            HIAPI.EventService.unfavoriteBy(name: event.name)
+            .onCompletion { result in
+                switch result {
+                case .success:
+                    DispatchQueue.main.async {
+                        HILocalNotificationController.shared.unscheduleNotification(for: event)
+                        event.favorite = false
+                        sender.setToggle(active: event.favorite)
```
Comment (rauhul) [inline HackIllinois/ViewControllers/HIEventDetailViewController.swift]: review this line

File: HackIllinois/UI/TableView/Cells/HIEventDetailLocationCell.swift [hunk 7]
```diff
@@ -13,6 +13,7 @@
 import Foundation
 import UIKit
 import MapKit
+import HIAPI
```
Comment (rauhul) [inline HackIllinois/UI/TableView/Cells/HIEventDetailLocationCell.swift]: why is this here?
